code/extensions/ShareDraftContentExtension.php

```diff
@@ -1,5 +1,78 @@
 <?php
 
 class ShareDraftContentExtension extends DataExtension {
+	/**
+	 * @var array
+	 */
+	private static $db = array(
+		'ShareTokenSalt' => 'Varchar(16)',
+	);
 
+	/**
+	 * @var array
+	 */
+	private static $has_many = array(
+		'ShareTokens' => 'ShareToken',
+	);
+
+	/**
+	 * @return string
+	 */
+	public function ShareTokenLink() {
+		$shareToken = $this->getNewShareToken();
+
+		return Controller::join_links($this->owner->Link(), 'preview', $this->generateKey($shareToken->Token), $shareToken->Token);
+	}
+
+	/**
+	 * @return ShareToken
+	 */
+	protected function getNewShareToken() {
+		if(!$this->owner->ShareTokenSalt) {
+			$this->owner->ShareTokenSalt = $this->getNewToken();
+			$this->owner->write();
+		}
+
+		$token = null;
+
+		while(true) {
```

Comment:
I'm scared. Can you have some kind of limit on this?

I also notice your first iteration of this loop tests `"Token" => null`. Is this intentional?